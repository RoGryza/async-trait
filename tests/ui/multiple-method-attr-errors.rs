use async_trait::async_trait;

#[async_trait(?Send)]
pub trait Trait {
    #[async_trait(?Send)]
    #[async_trait]
    #[async_trait(?Send)]
    async fn method();
}

#[async_trait]
pub trait Trait2 {
    #[async_trait(?Send)]
    #[async_trait(?Send)]
    async fn method();
}

fn main() {}
