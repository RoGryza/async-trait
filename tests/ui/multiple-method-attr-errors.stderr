error: duplicate #[async_trait] in method
 --> tests/ui/multiple-method-attr-errors.rs:6:5
  |
6 |     #[async_trait]
  |     ^^^^^^^^^^^^^^

error: duplicate #[async_trait] in method
 --> tests/ui/multiple-method-attr-errors.rs:7:5
  |
7 |     #[async_trait(?Send)]
  |     ^^^^^^^^^^^^^^^^^^^^^

error: redundant #[async_trait(?Send)] in method
 --> tests/ui/multiple-method-attr-errors.rs:5:5
  |
5 |     #[async_trait(?Send)]
  |     ^^^^^^^^^^^^^^^^^^^^^

error: duplicate #[async_trait] in method
  --> tests/ui/multiple-method-attr-errors.rs:14:5
   |
14 |     #[async_trait(?Send)]
   |     ^^^^^^^^^^^^^^^^^^^^^
